docs: stress the importance of secret removal (#20792)
juju/juju: docs/reference/secret.md

```{important}

> Charms that create secrets should _always_ handle the `secret-remove` event. Secret revisions, even if obsolete, remain until removed by the charm. Failing to do so will cause them to accumulate indefinitely.

Reviewer: suggestion: Rephrase to improve clarity:

```suggestion
Charms that create secrets should _always_ handle the `secret-remove` event. That is because secret revisions, even if obsolete, remain until removed by the charm; if a charm does not remove them, they accumulate indefinitely.
```
